File: mini_project_front/src/app/classes/appointment.ts
import { Worker } from './worker';
import { Customer } from './customer';

export enum AppointmentStatus {
    pending = 'pending',
    accepted = 'accepted',
    refused = 'refused',
    completed = 'completed'
}

export class Appointment {
    worker: Worker;
    customer: Customer;
    date: Date;
    location: string;
    statues: AppointmentStatus;
    finalPrice: number;
    description: string;
    constructor(
        worker: Worker,
        customer: Customer,
        date: Date,
        location: string,
        statues: AppointmentStatus,
        finalPrice: number,
        description: string

    ) {
        this.worker = worker;
        this.customer = customer;
        this.date = date;
        this.location = location;
        this.statues = statues;
        this.finalPrice = finalPrice;
        this.description = description;

    }
}

File: mini_project/src/main/java/com/example/mini_project/entities/Appointment.java
package com.example.mini_project.entities;

import jakarta.persistence.*;
import lombok.*;

import java.io.Serializable;
import java.time.LocalDate;
import java.time.LocalDateTime;
@NoArgsConstructor
@AllArgsConstructor
@Setter
@Getter
@ToString
@Entity
@Table(name = "appointments")
public class Appointment implements Serializable {
    private static final long serialVersionUID = 1L;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "worker_id")
    private Worker worker;

    @ManyToOne
    @JoinColumn(name = "customer_id")
    private Customer customer;

    @Column(name = "date")
    private LocalDate date;

    @Column(name = "location")
    private String location;

    private AppointmentStatues statues;

    private double finalPrice;

private String Description;
}
